File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
using namespace std;

int N;
int stack[10001];
int cur = -1;

int main() {
    string command;
    cin >> N;


    while (N--){
        cin >> command;

        if (command == "push"){
            cur++;
            int num;
            cin >> num;
            stack[cur] = num;
        } else if (command == "pop") {
            if (cur == -1) {
                cout << -1 << endl;
            } else {
                cout << stack[cur] << endl;
                cur--;
            }
        } else if (command == "size") {
            cout << cur + 1 << endl;
        } else if (command == "empty") {
            if (cur != -1) {
                cout << 0 << endl;
            } else {
                cout << 1 << endl;
            }
        } else if (command == "top") {
            if (cur != -1) {
                cout << stack[cur] << endl;
            } else {
                cout << -1 << endl;
            }
        }
    }


    return 0;
}

File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

n = int(input())

stack = []
cur = -1
while n:
    line = input().split()
    command = line[0]
    if len(line) > 1:
        num = int(line[1])

    if command == "push":
        stack.append(num)
        cur += 1

    elif command == "pop":
        if cur == -1:
            print(-1)
        else:
            print(stack[cur])
            stack.pop()
            cur -= 1

    elif command == "size":
        print(cur + 1)
    
    elif command == "empty":
        if cur != -1:
            print(0)
        else:
            print(1)

    elif command == "top":
        if cur != -1:
            print(stack[cur])
        else:
            print(-1)
    
    n -= 1